packages/nextjs/hooks/scaffold-eth/useScaffoldEventRead.ts

```diff
@@ -0,0 +1,124 @@
+import { ContractAbi, ContractName } from "~~/utils/scaffold-eth/contract";
+import { ExtractAbiEventNames } from "abitype";
+import { useProvider, useContract } from "wagmi";
+import { useDeployedContractInfo } from "~~/hooks/scaffold-eth";
+import { ethers } from "ethers";
+import { useEffect, useState } from 'react';
+
+/**
+ * @dev reads events from a deployed contract
+ * @param config - The config settings
+ * @param config.contractName - deployed contract name
+ * @param config.eventName - name of the event to listen for
+ * @param config.fromBlock - the block number to start reading events from (default: 1000 blocks before current block)
+ * @param config.filters - filters to be applied to the event (parameterName: value)
+ * @param config.blockData - if set to true it will return the block data for each event (default: false)
+ * @param config.transactionData - if set to true it will return the transaction data for each event (default: false)
+ * @param config.receiptData - if set to true it will return the receipt data for each event (default: false)
+ */
+export const useScaffoldEventRead = <
+  TContractName extends ContractName,
+  TEventName extends ExtractAbiEventNames<ContractAbi<TContractName>>,
+>({
+  contractName,
+  eventName,
+  fromBlock,
+  filters,
+  blockData,
+  transactionData,
+  receiptData,
+}: {
+  contractName: TContractName;
+  eventName: TEventName;
+  fromBlock?: number;
+  filters?: any;
+  blockData?: boolean;
+  transactionData?: boolean;
+  receiptData?: boolean;
+}) => {
+  const [events, setEvents] = useState(<any>[]);
+  const { data: deployedContractData } = useDeployedContractInfo(contractName);
+  const provider = useProvider();
+
+  const contract = useContract({
+    address: deployedContractData?.address,
+    abi: deployedContractData?.abi,
+    signerOrProvider: provider,
+  });
+
+  useEffect(() => {
+    async function readEvents() {
+      if (!contract) {
+        console.log("contract not found");
+        return;
+      }
+      try {
+        if (!fromBlock) {
+          fromBlock = await provider.getBlockNumber() - 1000;
```

Comment:
won't `fromBlock = 0` be a bit overkill default value 😅 ? Also, I think it will increase fetching time on chains other than a hardhat

Maybe We mention it in JSDoc comments that by default it starts filtering from the last 1000 blocks

Reply:
Yeah, true. Not sure what the best choice is. Maybe no default and make it explicit? I'm just not a fan of magic numbers in code, and 1000 feels a bit like it, could just as well be 50 or 100 or 500... So what about making the param mandatory? (just thinking out loud here :sweat_smile: )

Reply:
As @technophile-04 said, I didn't want to use fromBlock = 0 as default to avoid a long fetching time if you are using it on a test or mainnet.

Maybe we can set the fromBlock param mandatory, so the user has to think about what block number he needs to set and suggest that if you want all the emitted events, you should set the fromBlock equal to the block from the contract deployment.

Reply:
fromBlock is mandatory now!